### PR TITLE
cloud: convert uses of worspaces.operations into workspaces.executionMode

### PR DESCRIPTION
**Pull request by uturunku1**
## Description

The cloud package currently uses a deprecated API on workspaces to determine a workspace's execution mode. 

Deprecated:  Operations (boolean)
New hotness: Execution mode (string - "local", "remote", or "agent")

More details: https://www.terraform.io/docs/cloud/api/workspaces.html#request-body

All uses of Operations field coming from the client (within the cloud package) should be converted to the appropriate ExecutionMode equivalent.
Also, we need to update all acknowledgment of operations field on the tests that are testing the behavior of workspaces.

The usage of operations within `entilements` context is not being replaced because `type` entitlements does not have an ExecutionMode field.

Lastly, the usage of `Operations: *options.Operations` within `internal/cloud/tfe_client_mock.go` is also not being removed because is indirectly being called by the tests outside this package (specifically the tests within `internal/backend/remote`).

### CODE REVIEW

**Review by nfagerlund — COMMENTED** [review]
Huh, the repeated string comparisons make me feel a little weird. Admittedly there's only two of them in "real" code, and it's not like I'm _really_ expecting we'll introduce a fourth ExecutionMode... but still.

Do you think it might be a good idea to pull the knowledge about what the execution modes mean into a bool-returning helper function? That might just be paranoid of me.

I do have one piece of feedback I'm sure of, which is that at least one of those test cases down there should use `"agent"` instead of `"remote"` — Terraform CLI should treat those identically, so might as well codify that now that we're finally gonna start switching on ExecutionMode.

**Comment by uturunku1** [comment]
@nfagerlund Hey Nick! I made changes based on your suggestions here https://github.com/hashicorp/terraform/pull/29844#pullrequestreview-794479720 . Let me know if I misunderstood what you were suggesting. Thanks for your review!

File: internal/cloud/backend_apply_test.go [hunk 1]
```diff
@@ -1566,7 +1566,7 @@ func TestCloud_applyVersionCheck(t *testing.T) {
 				hasRemote := strings.Contains(output, "Running apply in Terraform Cloud")
 				hasSummary := strings.Contains(output, "1 added, 0 changed, 0 destroyed")
 				hasResources := run.State.HasManagedResourceInstanceObjects()
-				if !tc.forceLocal && tc.hasOperations {
+				if !tc.forceLocal && tc.executionMode == "remote" {
```
Comment (nfagerlund) [inline internal/cloud/backend_apply_test.go:1569]:
```suggestion
				if !tc.forceLocal && !isLocalExecutionMode(tc.executionMode) {
```

Whatcha think about using the same function here in the tests? Reason being: I think to be accurate, you'd want to make sure the `"agent"` execution mode also results in remote operations, and comparing to "remote" would miss that.

**Review by nfagerlund — APPROVED** [review]
This looks good to me! I have one more suggestion, but since it's in a test where we're controlling all the input right there in the table above anyway, it's not really a blocker.

File: internal/cloud/backend_apply_test.go (hunk 1 above)
Comment (uturunku1) [reply internal/cloud/backend_apply_test.go:1569]: Yes, this makes complete sense. I'll commit this suggestion :)

**Pull request merged**